### ISSUE TITLE
MySQL error

### ISSUE THREAD

**Issue by glooom** [savetheinternet/Tinyboard#136]
Hi. I'm using Ubuntu 12.04 on my server, and today, after installing several software updates (it was a long list), I've got an error during installation of Tinyboard: "Table 'newchan.boards' doesn't exist". 

I looked into the database after that: it contained only one table (posts_b) and 4 indexes -  id, thread_id, time, body. I restored previous version of OS, and everything became fine again - second installation was finished successfully. Is it a known problem with Ubuntu?

**Comment by infamousbutterly** [comment]
Don't create newchan... you will be viciously raped by nuchan.

**Comment by glooom** [comment]
newchan - is just a testname :) not a domain.

**Comment by savetheinternet** [comment]
Hm, okay. I am assuming you are using a MySQL server older than 5.5.3, and that is what is causing this issue. Can you confirm this?

Tinyboard still works with < 5.5.3, but it does things a little differently because the `utf8mb4` charset doesn't exist.

This is the code that I think is having problems (specifically the third line):

``` php
foreach ($queries as $query) {
    if ($mysql_version < 50503)
        $query = preg_replace('/(CHARSET=|CHARACTER SET )utf8mb4/', '$1utf8', $query);
    $query = preg_replace('/^([\w\s]*)`([0-9a-zA-Z$_\x{0080}-\x{FFFF}]+)`/u', '$1``$2``', $query);
    if (!query($query))
        $sql_errors .= '<li>' . db_error() . '</li>';
}
```

I've read over it about 5 times now and I still don't understand what could be causing this bug. Did this happen recently, or maybe a week or so ago? (And were you using the absolute latest version at the time?)

**Comment by savetheinternet** [comment]
It's strange that it was only able to insert the posts table. Did you get any SQL warnings/errors when installing Tinyboard?

**Comment by glooom** [comment]
Thanks for answer, savetheinternet. 
Currently I'm using MySQL version 5.5.32 and the latest version of Tinyboard from this site. I tried to install it about 8 hours ago and problem appeared only after ubuntu software updates.

I hadn't any other error messages. The only one was "Table 'newchan.boards' doesn't exist".

And here is the list of software updates (if it can help).

```
apache2-utils   utility programs for webservers     New version 2.2.22-1ubuntu1.4   Precise
apt     commandline package manager     New version 0.8.16~exp12ubuntu10.12     Precise
apt-utils   package managment related utility programs  New version 0.8.16~exp12ubuntu10.12     Precise
bind9-host  Version of 'host' bundled with BIND 9.X     New version 9.8.1.dfsg.P1-4ubuntu0.7    Precise
curl    Get a file from an HTTP, HTTPS or FTP server    New version 7.22.0-3ubuntu4.2   Precise
dnsutils    Clients provided with BIND  New version 9.8.1.dfsg.P1-4ubuntu0.7    Precise
firefox-locale-en   English language pack for Firefox   New version 23.0+build2-0ubuntu0.12.04.1    Precise
gnupg   GNU privacy guard - a free PGP replacement  New version 1.4.11-3ubuntu2.3   Precise
gpgv    GNU privacy guard - signature verification tool     New version 1.4.11-3ubuntu2.3   Precise
grub-common     GRand Unified Bootloader (common files)     New version 1.99-21ubuntu3.10   Precise
libapt-inst1.4  deb package format runtime library  New version 0.8.16~exp12ubuntu10.12     Precise
libapt-pkg4.12  package managment runtime library   New version 0.8.16~exp12ubuntu10.12     Precise
libbind9-80     BIND9 Shared Library used by BIND   New version 9.8.1.dfsg.P1-4ubuntu0.7    Precise
libcurl3    Multi-protocol file transfer library (OpenSSL)  New version 7.22.0-3ubuntu4.2   Precise
libdns81    DNS Shared Library used by BIND     New version 9.8.1.dfsg.P1-4ubuntu0.7    Precise
libdrm-intel1   Userspace interface to intel-specific kernel DRM services -- runtime    New version 2.4.43-0ubuntu0.0.2     Precise
libdrm-nouveau1a    Userspace interface to nouveau-specific kernel DRM services -- runtime  New version 2.4.43-0ubuntu0.0.2     Precise
libdrm-radeon1  Userspace interface to radeon-specific kernel DRM services -- runtime   New version 2.4.43-0ubuntu0.0.2     Precise
libdrm2     Userspace interface to kernel DRM services -- runtime   New version 2.4.43-0ubuntu0.0.2     Precise
libgcrypt11     LGPL Crypto library - runtime library   New version 1.5.0-3ubuntu0.2    Precise
libisc83    ISC Shared Library used by BIND     New version 9.8.1.dfsg.P1-4ubuntu0.7    Precise
libisccc80  Command Channel Library used by BIND    New version 9.8.1.dfsg.P1-4ubuntu0.7    Precise
libisccfg82     Config File Handling Library used by BIND   New version 9.8.1.dfsg.P1-4ubuntu0.7    Precise
libldap-2.4-2   OpenLDAP libraries  New version 2.4.28-1.1ubuntu4.3     Precise
liblwres80  Lightweight Resolver Library used by BIND   New version 9.8.1.dfsg.P1-4ubuntu0.7    Precise
libssl1.0.0     SSL shared libraries    New version 1.0.1-4ubuntu5.10   Precise
libudev0    udev library    New version 175-0ubuntu9.4  Precise
libxml2     GNOME XML library   New version 2.7.8.dfsg-5.1ubuntu4.6     Precise
linux-firmware  Firmware for Linux kernel drivers   New version 1.79.6  Precise
linux-image-virtual     Linux kernel image for virtual machines     New version 3.2.0.51.61     Precise
lsb-base    Linux Standard Base 4.0 init script functionality   New version 4.0-0ubuntu20.3     Precise
lsb-release     Linux Standard Base version reporting utility   New version 4.0-0ubuntu20.3     Precise
openssl     Secure Socket Layer (SSL) binary and related cryptographic tools    New version 1.0.1-4ubuntu5.10   Precise
udev    rule-based device node and kernel event manager     New version 175-0ubuntu9.4  Precise
webmin  web-based administration interface for Unix systems
```

**Comment by savetheinternet** [comment]
I think maybe your server is cluttered with open sores shitware. Try installing some proprietary software.

**Comment by Circlepuller** [comment]
I would think this is more of a MySQL related issue than a Tinyboard one. IMO go bother Oracle about this

**Comment by yuri410** [comment]
I had a similar issue today installing the board. I ended up executing install.sql manually to avoid this missing table issue. And then run install.php with step=5 to continue.

I also ran into a minor issue which at first I used a numerical database password the php treated it as a number. I guess this caused some formatting issues when generating instance-config.php (no quotes). Anyway I now use a better password.
